src/requirements/Farcaster/FarcasterRequirement.tsx

```diff
@@ -47,11 +47,24 @@ const FarcasterRequirement = (props: RequirementProps) => {
           case "FARCASTER_RECAST":
             return (
               <>
-                {`${requirement.type === "FARCASTER_LIKE" ? "Like" : "Recast"} the `}
-                <DataBlockWithCopy text={requirement.data.hash}>
-                  {shortenHex(requirement.data.hash, 3)}
-                </DataBlockWithCopy>
-                {` cast`}
+                {requirement.type === "FARCASTER_LIKE" ? "Like" : "Recast"}
+
+                {requirement.data?.hash && (
+                  <>
+                    {` the `}
+                    <DataBlockWithCopy text={requirement.data.hash}>
```

Comment:
Let's use a link here + I think in case the user provided a hash, we could fetch the cast info with the `https://api.neynar.com/v2/farcaster/cast/?api_key=...&viewer_fid=-1&type=hash&identifier=...` endpoint, and display a Warpcast link in that case too ;)